### PR TITLE
SDIT-4023 Resync prisoner when a movement is updated in the NOMIS admin screen

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementService.kt [hunk 1]
```diff
@@ -45,6 +47,7 @@ class CourtSchedulerSyncMovementService(
 
   suspend fun courtMovementChanged(event: ExternalMovementEvent) = when {
     event.movementType != CRT -> {}
+    event.auditModuleName == EDIT_EXTERNAL_MOVEMENTS_AUDIT_MODULE -> migrationService.resyncPrisonerCourtMovements(event.offenderIdDisplay!!)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementService.kt:50]: We can't recover if a user updates the movement sequence in the Edit External Movements admin screen. Therefore we call the prisoner "repair endpoint" in DPS that we would call after booking edge cases.

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -190,7 +190,7 @@ hmpps.sqs:
     eventexternalmovements:
       dlqMaxReceiveCount: 1
     eventcourtmovements:
-      dlqMaxReceiveCount: 1
+      dlqMaxReceiveCount: 2
```
Comment (mikehalmamoj) [inline src/test/resources/application-test.yml:193]: As we're not calling the resync endpoint in the usual way, I wanted to make sure that DLQ based retries to DPS and the mapping service work.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncScheduleIntTest.kt [hunk 3]
```diff
@@ -352,7 +353,7 @@ class CourtSchedulerSyncScheduleIntTest(
 
       @Test
       fun `should create failure telemetry`() {
-        verify(telemetryClient).trackEvent(
+        verify(telemetryClient, times(2)).trackEvent(
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncScheduleIntTest.kt:356]: A side effect of turning on a single retry in the DLQ.

**Review by steverendell — APPROVED** [review]
lgtm
